lib/config/path-util.js

```diff
@@ -0,0 +1,110 @@
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+'use strict';
+
+const path = require('path');
+
+module.exports = {
+    /**
+     * Determines the "contentBase" used for the devServer.
+     *
+     * @param {WebpackConfig} webpackConfig
+     * @return {String}
+     */
+    getContentBase(webpackConfig) {
+        // strip trailing slash (for Unix or Windows)
+        const outputPath = webpackConfig.outputPath.replace(/\/$/,'').replace(/\\$/, '');
+        // use the manifestKeyPrefix if available
+        const publicPath = webpackConfig.manifestKeyPrefix ? webpackConfig.manifestKeyPrefix.replace(/\/$/,'') : webpackConfig.publicPath.replace(/\/$/,'');
+
+        /*
+         * We use the intersection of the publicPath (or manifestKeyPrefix) and outputPath
+         * to determine the "document root" of the web server. For example:
+         *   * outputPath = /var/www/public/build
+         *   * publicPath = /build/
```

Comment:
You're correct. But, if the user has a `publicPath` that is not a sub-string of the `outputPath`, they will already have received an error forcing them to set `manifestKeyPrefix` (you do not need to set this manually... until you deploy under a subdirectory or set publicPath to a CDN). So, this situation won't happen. I've just pushed a commit to hopefully clarify this - it's a bit complex since different parts of the system are working together. It *is* possible that the user will set a `manifestKeyPrefix` that is not a substring of the `outputPath`, and then they would see this error (because this is not allowed)